app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -40,38 +46,63 @@ constructor(
     }
   }
 
-  fun getUpcomingEvents(uid: String) {
+  fun getUpcomingEvents() {
     viewModelScope.launch {
-      when (val userResponse = userRepository.getUser(uid)) {
-        is Resource.Success -> {
-          val user = userResponse.data!!
-          val attendeeList = user.eventsAttendeeSet.toList()
-          if (attendeeList.isNotEmpty()) {
-            when (val events = eventRepository.getEventsByIds(attendeeList)) {
-              is Resource.Success -> {
-                _uiState.value =
-                    _uiState.value.copy(
-                        eventList =
-                            EventList(
-                                events.data, events.data, _uiState.value.eventList.selectedEvent))
-              }
-              is Resource.Failure -> {
-                Log.d("EventsOverviewViewModel", "Error getting events for $uid")
-                _uiState.value =
-                    _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
-              }
-            }
-          } else {
+      userRepository.getCurrentUserId().let { userIdResource ->
```

Comment:
have you thought of perhaps putting this in a some sort of loop to treaty fetching the events? perhaps firebase bugs at the instance u attempted, I personally think it is worth reattempting.

Reply:
I believe this works fine. Don't think changing this would be time-efficient.